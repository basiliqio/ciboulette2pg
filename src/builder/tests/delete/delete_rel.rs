use super::*;

#[test]
fn simple_single_rel() {
    let ciboulette_store = gen_bag();
    let table_store = gen_table_store();
    let parsed_url =
		Url::parse("http://localhost/peoples/6720877a-e27e-4e9e-9ac0-3fff4deb55f2/relationships/favorite_color").unwrap();
    const INTENTION: CibouletteIntention = CibouletteIntention::Delete;

    let req_builder = CibouletteRequestBuilder::new(INTENTION, &parsed_url, &None);
    let request = req_builder.build(&ciboulette_store).unwrap();
    let ciboulette_request = CibouletteDeleteRequest::try_from(request).unwrap();
    let mut builder = Ciboulette2PostgresBuilder::default();
    builder
        .gen_delete(&ciboulette_store, &table_store, &ciboulette_request)
        .unwrap();
    let res = builder.build().unwrap();

    test_sql!(res);
}

#[test]
fn simple_single_rel_non_optional() {
    let ciboulette_store = gen_bag();
    let table_store = gen_table_store();
    let parsed_url = Url::parse(
        "http://localhost/comments/6720877a-e27e-4e9e-9ac0-3fff4deb55f2/relationships/author",
    )
    .unwrap();
    const INTENTION: CibouletteIntention = CibouletteIntention::Delete;

    let req_builder = CibouletteRequestBuilder::new(INTENTION, &parsed_url, &None);
    let request = req_builder.build(&ciboulette_store).unwrap();
    let ciboulette_request = CibouletteDeleteRequest::try_from(request).unwrap();
    let mut builder = Ciboulette2PostgresBuilder::default();
    let err = builder
        .gen_delete(&ciboulette_store, &table_store, &ciboulette_request)
        .unwrap_err();
    assert_eq!(
        matches!(err, Ciboulette2SqlError::RequiredRelationship(x, y) if x == "comments" && y == "author_id"),
        true
    );
}

#[test]
fn multi_relationships() {
    let ciboulette_store = gen_bag();
    let table_store = gen_table_store();
    let parsed_url = Url::parse(
        "http://localhost/peoples/6720877a-e27e-4e9e-9ac0-3fff4deb55f2/relationships/articles",
    )
    .unwrap();
    const INTENTION: CibouletteIntention = CibouletteIntention::Delete;

    let req_builder = CibouletteRequestBuilder::new(INTENTION, &parsed_url, &None);
    let request = req_builder.build(&ciboulette_store).unwrap();
    let ciboulette_request = CibouletteDeleteRequest::try_from(request).unwrap();
    let mut builder = Ciboulette2PostgresBuilder::default();
    let err = builder
        .gen_delete(&ciboulette_store, &table_store, &ciboulette_request)
        .unwrap_err();
    assert_eq!(
        matches!(err, Ciboulette2SqlError::BulkRelationshipDelete),
        true
    );
}
