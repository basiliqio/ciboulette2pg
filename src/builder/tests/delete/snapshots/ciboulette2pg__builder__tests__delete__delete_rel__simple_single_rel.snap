---
source: src/builder/tests/delete/delete_rel.rs
expression: "sqlformat::format(query.as_str(), &sqlformat::QueryParams::None,\n                  sqlformat::FormatOptions::default())"

---
UPDATE
  "public"."peoples"
SET
  "favorite_color" = NULL
WHERE
  "public"."peoples"."id" = $1;
